### PR TITLE
kernel: enable memory hotplug

### LINKED ISSUE #262: kernel: enable memory hotplug
memory hot plug is a feature required in Clear Containers

### PR DESCRIPTION
**Pull request by devimc**
memory hotplug is a feature needed in Clear Containers,
with this patch hot-added memory is connected by the kernel
automatically

fixes #262

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by jcvenegas** [comment]
lgtm

**Comment by jcvenegas** [comment]
@grahamwhaley  please take a look

**Review by jcvenegas — APPROVED** [review]

File: kernel/kernel-config-4.14.x [hunk 1]
```diff
@@ -505,8 +510,10 @@ CONFIG_ARCH_WANTS_THP_SWAP=y
 # CONFIG_ZSMALLOC is not set
 CONFIG_GENERIC_EARLY_IOREMAP=y
 CONFIG_ARCH_SUPPORTS_DEFERRED_STRUCT_PAGE_INIT=y
+# CONFIG_DEFERRED_STRUCT_PAGE_INIT is not set
```
Comment (jodh-intel) [inline kernel/kernel-config-4.14.x:513]: Are these new disabled options "magic" ones that appeared as a result of enabling memory hotplug, or are they due to a new kernel version?

Wait, did I just spot a `CONFIG_OMG_THIS_ENTIRE_LINUX_KERNEL_CONFIG_THING_HAS_BECOME_ONE_GIGANTIC_HAIR_BALL_OF_CONFUSING_AND_MAGIC_OPTIONS_WHICH_NOBODY_WITHOUT_UBER_NINJA_SKILLS_REALLY_UNDERSTANDS=y`. ... Hmm... maybe not. :smile:

Comment (jcvenegas) [reply kernel/kernel-config-4.14.x:513]: @jodh-intel  yes it is sometimes is difficult to know what  options were enabled and what of the rest were automatically added.  
A good point to add PR process for kata. In the commit message desrcribe what kernel config was added explicitly.

Comment (jodh-intel) [reply kernel/kernel-config-4.14.x:513]: Yep. But to be clear, I'm not criticizing this PR, I'm moaning about the general way the kernel is configured as it's too complex and "too magic" imho ;)

Comment (jcvenegas) [reply kernel/kernel-config-4.14.x:513]: @jodh-intel  yep agree :P

File: kernel/kernel-config-4.14.x [hunk 2]
```diff
@@ -2209,6 +2218,7 @@ CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_TMPFS_XATTR=y
 CONFIG_HUGETLBFS=y
 CONFIG_HUGETLB_PAGE=y
+CONFIG_ARCH_HAS_GIGANTIC_PAGE=y
```
Comment (grahamwhaley) [inline kernel/kernel-config-4.14.x:2221]: It's not clear to me if all of these are related to enabling the hotplug memory or not (as noted by @jodh-intel ). @devimc , can you tell us if some of these are just kernel version bump related or not?
Also, have we stared hard at all the items that got turned on, and ensured that we really do need/want them turned on for our container kernel. Previously we've had things get turned on as the default, but they have not been needed by us. Thus, I'd like to ensure each newly enabled item has been checked for validity (or we may slowly bloom the kernel with things we don't actually need).

Yes, kernel config is complex and inter-twined. There have been efforts to make it easier and more understandable, but we are still with what we have - it does the job, and fixing it is a big hard problem...

**Pull request merged**